src/frameworks/compose/discover/runtime/node.ts

```diff
@@ -0,0 +1,276 @@
+    if (!engine) {
+      const latest = VALID_ENGINES.node[VALID_ENGINES.node.length - 1];
+      return `node:${latest}-slim`;
+    }
+    const versionNumber = parseInt(engine.node, 10);
+    const validEngines = VALID_ENGINES.node.filter((it) => it !== versionNumber);
+
+    if (!validEngines.length) {
+      throw new FirebaseError(
+        `This integration expects Node version ${conjoinOptions(
+          VALID_ENGINES.node,
+          "or"
+        )}. You're running version ${versionNumber}, which is not compatible.`
+      );
+    }
+
+    return `node:${versionNumber}-slim`;
+  }
+
+  async getPackageManager(fs: FileSystem): Promise<string> {
+    if (await fs.exists(YARN_LOCK)) {
+      return YARN;
+    }
+
+    return NPM;
+  }
+
+  async getDependenciesForNPM(
+    fs: FileSystem,
+    packageJSON: PackageJSON
+  ): Promise<Record<string, string>> {
+    const directDependencies = { ...packageJSON.dependencies, ...packageJSON.devDependencies };
+    let transitiveDependencies = {};
+
+    const packageLockJSONRaw = await readOrNull(fs, PACKAGE_LOCK_JSON);
+    if (!packageLockJSONRaw) {
+      return directDependencies;
+    }
+    const packageLockJSON = JSON.parse(packageLockJSONRaw);
+    const directDependencyNames = Object.keys(directDependencies).map((x) =>
+      join("node_modules", x)
+    );
+
+    directDependencyNames.forEach((directDepName) => {
+      const transitiveDeps = packageLockJSON.packages[directDepName].dependencies;
+      transitiveDependencies = { ...transitiveDependencies, ...transitiveDeps };
+    });
+
+    return { ...directDependencies, ...transitiveDependencies };
+  }
+
+  async getDependenciesForYARN(
+    fs: FileSystem,
+    packageJSON: PackageJSON
+  ): Promise<Record<string, string>> {
+    const directDependencies = { ...packageJSON.dependencies, ...packageJSON.devDependencies };
+    const yarnLockJSONRaw = await readOrNull(fs, YARN_LOCK);
+    if (!yarnLockJSONRaw) {
+      return directDependencies;
+    }
+
+    const allDependencies: any = {};
+    const lines = yarnLockJSONRaw.split("\n");
+
+    for (let line of lines) {
+      line = line.trim();
+      if (line.startsWith("#") || line === "") {
+        continue;
+      }
+      const patternMatch = /^"(.+?)@(.+?)":/.exec(line);
+      if (patternMatch) {
+        const dependencyName = patternMatch[1];
+        const version = patternMatch[2];
+        allDependencies[dependencyName] = version;
+      }
+    }
+
+    return allDependencies;
+  }
+
+  async getDependencies(
+    fs: FileSystem,
+    packageJSON: PackageJSON,
+    packageManager: string
+  ): Promise<Record<string, string>> {
+    try {
+      let dependencies = {};
+      if (packageManager === NPM) {
+        dependencies = await this.getDependenciesForNPM(fs, packageJSON);
+      } else if (packageManager === YARN) {
+        dependencies = await this.getDependenciesForYARN(fs, packageJSON);
+      }
```

Comment:
suggestion - let's just get deps from package.json, regardless of package manager.

Reply:
Done